### PR TITLE
Fix formatParameter() for floats

### LINKED ISSUE #1373: TypeError - migrations fail after upgrade to 3.7.0
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 3.7.0

#### Summary

when running my set of migrations against an empty database with 3.6.0 everything works, with 3.7.0 it fails

#### Previous behavior

migrations should run

#### Current behavior

```
[error] Migration Migrations\Version20180326141405 failed during Execution. Error: "Doctrine\Migrations\InlineParameterFormatter::formatParameter(): Return value must be of type string|int|null, float returned"
PHP Fatal error:  Uncaught TypeError: Doctrine\Migrations\InlineParameterFormatter::formatParameter(): Return value must be of type string|int|null, float returned in /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/InlineParameterFormatter.php:65
Stack trace:
#0 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/InlineParameterFormatter.php(45): Doctrine\Migrations\InlineParameterFormatter->formatParameter()
#1 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(288): Doctrine\Migrations\InlineParameterFormatter->formatParameters()
#2 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(269): Doctrine\Migrations\Version\DbalExecutor->outputSqlQuery()
#3 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(150): Doctrine\Migrations\Version\DbalExecutor->executeResult()
#4 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/Version/DbalExecutor.php(72): Doctrine\Migrations\Version\DbalExecutor->executeMigration()
#5 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(88): Doctrine\Migrations\Version\DbalExecutor->execute()
#6 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(54): Doctrine\Migrations\DbalMigrator->executePlan()
#7 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/DbalMigrator.php(136): Doctrine\Migrations\DbalMigrator->executeMigrations()
#8 /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php(212): Doctrine\Migrations\DbalMigrator->migrate()
#9 /project/vendor/symfony/console/Command/Command.php(326): Doctrine\Migrations\Tools\Console\Command\MigrateCommand->execute()
#10 /project/vendor/symfony/console/Application.php(1063): Symfony\Component\Console\Command\Command->run()
#11 /project/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
#12 /project/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
#13 /project/vendor/doctrine/doctrine-module/bin/doctrine-module.php(45): Symfony\Component\Console\Application->run()
#14 /project/vendor/doctrine/doctrine-module/bin/doctrine-module(4): include('...')
#15 /project/vendor/bin/doctrine-module(119): include('...')
#16 {main}
  thrown in /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/InlineParameterFormatter.php on line 65

Fatal error: Uncaught TypeError: Doctrine\Migrations\InlineParameterFormatter::formatParameter(): Return value must be of type string|int|null, float returned in /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/InlineParameterFormatter.php on line 65

TypeError: Doctrine\Migrations\InlineParameterFormatter::formatParameter(): Return value must be of type string|int|null, float returned in /project/vendor/doctrine/migrations/lib/Doctrine/Migrations/InlineParameterFormatter.php on line 65
```

#### How to reproduce

```php
 private function formatParameter(mixed $value, string|int $type): string|int|null
    {
        if (is_string($type) && Type::hasType($type)) {
            $convertToDatabaseValue = Type::getType($type)->convertToDatabaseValue(
                $value,
                $this->connection->getDatabasePlatform(),
            );
            return $convertToDatabaseValue;
        }

        return $this->parameterToString($value);
    }
```

`$value` is a double of `0.5`
`$type` is a string with `string`
and database platform is "SQLite" on PHP 8.2.10

### PR DESCRIPTION
**Pull request by derrabus**
|      Q       |   A
|------------- | -----------
| Type         | bug
| BC Break     | no
| Fixed issues | Fix #1373

### CODE REVIEW

File: lib/Doctrine/Migrations/InlineParameterFormatter.php [hunk 1]
```diff
@@ -64,14 +65,14 @@ private function formatParameter(mixed $value, string|int $type): string|int|nul
...
+    /** @param int[]|bool[]|string[]|float[]|array|int|string|float|bool $value */
+    private function parameterToString(array|int|string|float|bool $value): string
     {
         if (is_array($value)) {
             return implode(', ', array_map($this->parameterToString(...), $value));
         }
 
-        if (is_int($value) || is_string($value)) {
+        if (is_int($value) || is_string($value) || is_float($value)) {
             return (string) $value;
```
Comment (SenseException) [inline lib/Doctrine/Migrations/InlineParameterFormatter.php:76]: Why did a string value need a string type cast?

**Review by SenseException — APPROVED** [review]
You already asked it in the issue, but I'd also like to know if this will fix @BigMichi1's problem.

File: lib/Doctrine/Migrations/InlineParameterFormatter.php (hunk 1 above)
Comment (derrabus) [reply lib/Doctrine/Migrations/InlineParameterFormatter.php:76]: It doesn't, it's a cheap no-op. I suppose the original author simply wanted to keep the code short.

**Comment by BigMichi1** [comment]
For me this works now

**Review by greg0ire — APPROVED** [review]

**Pull request merged**